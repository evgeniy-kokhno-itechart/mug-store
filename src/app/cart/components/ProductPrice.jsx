import React from 'react';
import { PropTypes } from 'prop-types';
import { Spinner } from '../../shared';

const ProductPrice = ({ price, isCurrencyLoading }) => (isCurrencyLoading
  ? <Spinner spinnerClasses="spinner--small" />
  : <span>{price || 'ERROR!'}</span>);

export default ProductPrice;

ProductPrice.propTypes = {
  price: PropTypes.number.isRequired,
  isCurrencyLoading: PropTypes.bool,
};

ProductPrice.defaultProps = {
  isCurrencyLoading: false,
};
