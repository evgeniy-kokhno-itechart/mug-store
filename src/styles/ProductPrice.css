.product-price__spinner {
  width: 1rem;
  height: 1rem;
}
