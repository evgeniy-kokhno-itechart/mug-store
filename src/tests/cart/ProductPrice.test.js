/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ProductPrice from '../../cart/components/ProductPrice';

configure({ adapter: new Adapter() });

describe('<ProductPrice />', () => {
  let ProductPriceWrapper;

  beforeEach(() => {
    ProductPriceWrapper = shallow(<ProductPrice price={100} />);
  });

  it('check render without isCurrencyLoading provided', () => {
    expect(ProductPriceWrapper).toMatchSnapshot();
  });

  it('check render with isCurrencyLoading = true provided', () => {
    ProductPriceWrapper.setProps({ isCurrencyLoading: true });
    expect(ProductPriceWrapper).toMatchSnapshot();
  });
});
