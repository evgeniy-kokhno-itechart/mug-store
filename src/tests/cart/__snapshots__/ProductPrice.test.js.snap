// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductPrice /> check render with isCurrencyLoading = true provided 1`] = `
<Spinner
  spinnerClasses="spinner--small"
  wrapperClasses=""
/>
`;

exports[`<ProductPrice /> check render without isCurrencyLoading provided 1`] = `
<span>
  100
</span>
`;
